modules/docs/markdown/05-design/02-schemas.md

```diff
@@ -306,4 +306,118 @@ Regarding the `underlyingSchema` value in the companion object of `IntList`, you
 
 ### Enumerations
 
-**TODO** (waiting for smithy 2.0 which changes the syntax)
+Smithy allows for two types of enumerations : string and integer enumerations.
+
+Additionally, smithy4s supports specifying whether an operation is open or closed. An open enumeration allows
+for holding unknown values, whereas a closed one is strictly limited to a set of specified values. This brings
+the total number of possible "flavours" of enumerations to 4, which is reified via a `smithy4s.schema.EnumTag` ADT
+that comprises 4 different cases : one for each combination between `[open, closed]` and `[int, string]`.
+
+Enumerations are typically modelled as Algebraic Data types. Each case of an enumeration is associated with both a String and Int value. In the case of `intEnum`, the string value is the name of the case. In the case of a normal (string) `enum`, the integer value is the index of the case in the list.
+
+Additionally, each enumeration case holds its own hints.
+
+#### Closed enumerations
+
+Given this smithy code :
+
+```kotlin
+namespace example
+
+intEnum Numbers {
+  ONE = 1
+  TWO = 2
+}
+```
+
+The corresponding generated Scala-code is :
```

Comment:
did you use https://morning-bird-7081.fly.dev/ to preview that? trololol

Reply:
THE HELL IS THIS and how did I not know about it

Reply:
@daddykotex I'm curious as well : who built this and how did you become aware of it ?

Reply:
I built it to try Scala JS. I wanted to see what made a scala-js asset so big. You can see the increase over the different PR, so it's not perfect but it can give an idea: https://github.com/daddykotex/smithy4s-code-generation/pulls?q=is%3Apr+is%3Aclosed

To me it just prove that while it's nice to work with, I'd never ship a scala js in a production app meant to reach tons of user. For internal software I guess it can be fine.

The source is over at: https://github.com/daddykotex/smithy4s-code-generation/ 

Some ideas that I'll implement in the future (maybe):

- there should be a link to the source
- I've been working on it recently to allow the loading of external jars (not free form, but from a list of configured artifacts that would be made available on startup of the app, and added/removed to the codegen classpath on the fly depending on what the users select)
- download of a project zip with everything wired, mock service interface included

Reply:
Yeah. I mean I'm pretty sure that with a few tweaks we could have a fetch-based client easily, thus getting rid of the typelevel stack altogether. 

Alternatively I think if you're gonna use typelevel on a scala-js app, may as well use [calico](https://www.armanbilge.com/calico/) instead of laminar. Laminar has its own reactive constructs that overlap with what fs2 does. 

Anyhow, very fascinating stuff, thank you !